document-store/src/main/java/org/hypertrace/core/documentstore/postgres/utils/PostgresUtils.java

```diff
@@ -450,25 +455,31 @@ public static String prepareParsedNonCompositeFilter(
       case "NOT_IN":
       case "NOT IN":
         // NOTE: Pl. refer this in non-parsed expression for limitation of this filter
+        // In order to make the behaviour same as for mongo, the not in operator would match only if
+        // the lhs
+        // and rhs have no intersection at all
+        // NOTE: This doesn't work in case the lhs is a function
```

Comment:
Nit:
```suggestion
        // In order to make the behaviour same as for Mongo, the "NOT_IN" operator would match only if
        // the LHS and RHS have no intersection at all
        // NOTE: This doesn't work in case the LHS is a function
```